source/expressions.tex

```diff
@@ -4163,13 +4163,12 @@
 \pnum
 \indextext{expression!unary operator}%
 \indextext{operator!unary}%
-The unary \tcode{*} operator performs \defn{indirection}:
+The unary \tcode{*} operator performs \defn{indirection}.
 \indextext{dereferencing|see{indirection}}%
-the expression to which it is applied shall be a pointer to an object
-type, or a pointer to a function type and the result is an lvalue
-referring to the object or function to which the expression points. If
-the type of the expression is ``pointer to \tcode{T}'', the type of the
-result is ``\tcode{T}''.
+Its operand shall be a prvalue of type ``pointer to \tcode{T}'',
```

Comment:
@jensmaurer: is it a normative change to require that the operand be a prvalue, or was that previously implied by other wording?

Reply:
The descriptions of most operators are unclear whether their operands are prvalues or lvalues.
Anyway, for the dereferencing operator, it's pelucidly clear that it takes a prvalue, always. Maybe the prior text saying "shall be a pointer to an object" could be interpreted as "shall be a pointer value...:", meaning a prvalue of pointer type.

Reply:
I suppose this is largely inherited from C, where things aren't lvalues unless they're explicitly called out as lvalues.

The normative impact/question is whether there's an lvalue-to-rvalue conversion in something like `*p`. I agree that this can't really be anything else (after all, you need to "load" the pointer value before you can dereference it), but it just feels a tiny bit bold to establish such clarity editorially. Let's proceed optimistically, but let CWG know via email.

Reply:
@tkoeppe In general, it is CWG1642.

Reply:
@languagelawyer: thank you!